tests/core/test_observe.py

```diff
@@ -86,7 +86,7 @@ async def tick():
 
     async def watch():
         async for val in observe_value(sig, done_timeout=0.2):
-            time.sleep(0.15)
+            await asyncio.sleep(0.1)
```

Comment:
Ok, my fault for not putting a comment in here, but this really was meant to be a `time.sleep`! This is a test to prove a subtle timing bug where the inner loop of observe_value was blocking the event loop. Please can you restore this and put a comment to this effect in?

Reply:
Made a [new issue](https://github.com/bluesky/ophyd-async/pull/692), easier than a revert.